codegen/tests/simple/simple_gencallback.go

```diff
@@ -41,3 +37,21 @@ func (c Callback) AddMiddleware(ctx context.Context, r chi.Router) {
 func (c Callback) BasePath() string {
 	return "/"
 }
+
+func (c CallbackWithMapError) MapError(ctx context.Context, cause error) *common.HTTPError {
```

Comment:
`MapError` should take a `ServerError`, not just an `error` to make this more explicit.

You can modify `CreateError` to return a `ServerError` as follows:

```go
func CreateError(ctx context.Context, kind Kind, message string, cause error) ServerError {
	// we may push the error to NR here
	if err := CheckContextTimeout(ctx, message, cause); err != nil {
		return *err
	}

	e, ok := cause.(*ServerError)
	if !ok {
		return ServerError{Kind: kind, Message: message, Cause: cause}
	}

	return *e
}

func CheckContextTimeout(ctx context.Context, message string, cause error) *ServerError {
	if ctx.Err() == context.DeadlineExceeded {
		return &ServerError{Kind: DownstreamTimeoutError, Message: message, Cause: cause}
	}
	return nil
}
```

Reply:
This CreateError is referenced by other places as well those requires a generic error, do we need to create separate CreateServerError method instead?

Reply:
Ok. Leave it for now.